e2etest/bgptests/unnummbered.go

```diff
@@ -0,0 +1,198 @@
+
+	ginkgo.AfterEach(func() {
+		if ginkgo.CurrentSpecReport().Failed() {
+			k8s.DumpInfo(Reporter, ginkgo.CurrentSpecReport().LeafNodeText)
+			dumpBGPPeers(ReportPath, ginkgo.CurrentSpecReport().LeafNodeText, []*frrcontainer.FRR{externalP2PContainer})
+		}
+	})
+
+	ginkgo.Context("FRR when applying unnumbered config", func() {
+		ginkgo.It("Sessions should be established", func() {
+			rc := frrconfig.RouterConfigUnnumbered{
+				ASNLocal:      metalLBASN,
+				ASNRemote:     metalLBASN,
+				Hostname:      "tor1",
+				Interface:     "net1",
+				RouterID:      "1.1.1.1",
+				ToAdvertiseV4: []string{"1.1.1.0/24"},
+				ToAdvertiseV6: []string{"1111::/64"},
+			}
+
+			var err error
+			iface := rc.Interface
+			ginkgo.By(fmt.Sprintf("creating p2p %s:%s -- %s:external-p2p-container", nodeWithP2PConnection.Name, iface, iface))
+			externalP2PContainer, err = frrcontainer.CreateP2PPeerFor(nodeWithP2PConnection.Name, iface, FRRImage)
+			Expect(err).NotTo(HaveOccurred())
+			ginkgo.DeferCleanup(func() {
+				err := frrcontainer.Delete([]*frrcontainer.FRR{externalP2PContainer})
+				Expect(err).NotTo(HaveOccurred())
+			})
+			ginkgo.By(fmt.Sprintf("updating frrconfig to %s", externalP2PContainer.Name))
+			c, err := rc.Config()
+			Expect(err).NotTo(HaveOccurred())
+			err = externalP2PContainer.UpdateBGPConfigFile(c)
+			Expect(err).NotTo(HaveOccurred())
+
+			resources := config.Resources{
+				Peers: []metallbv1beta2.BGPPeer{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "tor",
+							Namespace: metallb.Namespace,
+						},
+						Spec: metallbv1beta2.BGPPeerSpec{
+							Interface: iface,
+							ASN:       rc.ASNRemote,
+							MyASN:     rc.ASNLocal,
+						},
+					},
+				},
+				Pools: []metallbv1beta1.IPAddressPool{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "bgp-test",
+						},
+						Spec: metallbv1beta1.IPAddressPoolSpec{
+							Addresses: prefixSend,
+						},
+					},
+				},
+				BGPAdvs: []metallbv1beta1.BGPAdvertisement{
+					{ObjectMeta: metav1.ObjectMeta{Name: "empty"}},
+				},
+			}
+
+			err = ConfigUpdater.Update(resources)
+			Expect(err).NotTo(HaveOccurred(), "apply the CR in k8s api failed")
+
+			nodeP2PContainer := executor.ForContainer(nodeWithP2PConnection.Name)
+			nodeLLA, err := netdev.LinkLocalAddressForDevice(nodeP2PContainer, iface)
+			Expect(err).NotTo(HaveOccurred())
+			// peerLLA, err := netdev.LinkLocalAddressForDevice(externalP2PContainer, iface)
+			// Expect(err).NotTo(HaveOccurred())
+
+			ginkgo.By("validating the node and p2p container peered")
+			validateUnnumberedBGPPeering(externalP2PContainer, nodeLLA)
+
+			svc, _ := testservice.CreateWithBackend(cs, testNamespace, "external-local-lb")
+			ginkgo.By("Checking the service gets an ip assigned")
+			for _, i := range svc.Status.LoadBalancer.Ingress {
+				ingressIP := jigservice.GetIngressPoint(&i)
+				err = config.ValidateIPInRange(resources.Pools, ingressIP)
+				Expect(err).NotTo(HaveOccurred())
+			}
```

Comment:
let's verify that the service got the prefixSend (it makes the test clearer)

Reply:
Given that prefixSend includes mask `/32, /128`, doing that check the result does not look cleaner.* IMO we can let that because is only even do not have check as the real check is following. I take another look once I have the per stack test

Reply:
Added the check we can keep that if you like it more (or revert)